mermithid/misc/FakeTritiumDataFunctions.py

```diff
@@ -0,0 +1,655 @@
+#No detailed lineshape option
+def convolved_spectral_rate(K, Q, mnu, Kmin,
+                            lineshape, ls_params, min_energy, max_energy):
+    #Integral bounds chosen to approximately minimize computation error
+    pts = []
+    lbound = K+min_energy
+    ubound = K+max_energy
+    if lbound<Q-mnu<ubound:
+        pts.append(Q-mnu)
+    if K>Kmin:
+        if lineshape=='gaussian':
+            return integrate.quad(_S_integrand_gaussian, lbound, ubound, args=(K, Q, mnu, ls_params[0]), epsrel=1e-2, epsabs=0, limit=150, points=pts)[0]
+        elif lineshape=='simplified_scattering' or lineshape=='simplified':
+            return integrate.quad(_S_integrand_simp, lbound, ubound, args=(K, Q, mnu, ls_params), epsrel=1e-2, epsabs=0, limit=150, points=pts)[0]
+    else:
+        return 0.
+
+
+#Convolution of background and lineshape by integrating directly
+#No detailed lineshape option
+#Currently, the whole background is smeared. In reality, only background from electons should be smeared; false events from noise fluctuations produce an adititional unsmeared background.
+def convolved_bkgd_rate(K, Kmin, Kmax, lineshape, ls_params, min_energy, max_energy):
+    #Integral bounds chosen to approximately minimize computation error
+    pts = []
+    lbound = K+min_energy
+    ubound = K+max_energy
+    if K>Kmin:
+        if lineshape=='gaussian':
+            return integrate.quad(_B_integrand_gaussian, lbound, ubound, args=(K, ls_params[0]), epsrel=1e-3, epsabs=0, limit=150, points=pts)[0]
+        elif lineshape=='simplified_scattering' or lineshape=='simplified':
+            return integrate.quad(_B_integrand_simp, lbound, ubound, args=(K, ls_params), epsrel=1e-3, epsabs=0, limit=150, points=pts)[0]
+    else:
+        return 0.
+
+
+#Convolution of signal and lineshape using scipy.signal.convolve
+def convolved_spectral_rate_arrays(K, Q, mnu, Kmin,
+                                   lineshape, ls_params, min_energy, max_energy):
+    """K is an array-like object
+    """
+    logger.info('Using scipy convolve')
+    energy_half_range = max(max_energy, abs(min_energy))
+
+    dE = K[1] - K[0]
+    n_dE_pos = round(energy_half_range/dE) #Number of steps for the lineshape for energies > 0
+    n_dE_neg = round(energy_half_range/dE) #Same, for energies < 0
+    K_lineshape =  np.arange(-n_dE_neg*dE, n_dE_pos*dE, dE)
+
+    #Generating finely spaced points on the lineshape
+    if lineshape=='gaussian':
+        logger.info('broadening: {}'.format(ls_params[0]))
+        lineshape_rates = gaussian(K_lineshape, [ls_params[0], 0])
+    elif lineshape=='simplified_scattering' or lineshape=='simplified':
+        lineshape_rates = simplified_ls(K_lineshape, 0, ls_params[0], ls_params[1], ls_params[2], ls_params[3], ls_params[4], ls_params[5])
+    elif lineshape=='detailed_scattering' or lineshape=='detailed':
+        lineshape_rates = spectrum_func(K_lineshape/1000., ls_params[0], 0, ls_params[1], 1)
+
```

Comment:
What if lineshape is not gaussian, simplified scattering or detailed scattering? Should the program run or report an error?

Reply:
it does now.